File: clients/src/main/java/me/zixuan/clients/ymlconfig/service/KafkaConsumerService.java
package me.zixuan.clients.ymlconfig.service;

import lombok.extern.log4j.Log4j2;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
@Log4j2
public class KafkaConsumerService {
    @KafkaListener(topics = "${spring.kafka.template.default-topic}")
    public void consume(String message) {
        log.info(String.format("Received message from Kafka topic: %s", message));
    }
}

File: clients-kt/src/main/kotlin/me/zixuan/clientskt/ymlconfig/service/KafkaConsumerService.kt
package me.zixuan.clientskt.ymlconfig.service

import org.slf4j.LoggerFactory
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service

@Service
class KafkaConsumerService {
    private val logger = LoggerFactory.getLogger(javaClass)
    @KafkaListener(topics = ["\${spring.kafka.template.default-topic}"])
    fun consume(message: String) {
        logger.info("Received message from Kafka topic: {}", message)
    }
}